ffi/src/scan.rs

```diff
@@ -349,26 +349,19 @@ fn row_indexes_from_dv_impl(
 
 // Wrapper function that gets called by the kernel, transforms the arguments to make the ffi-able,
 // and then calls the ffi specified callback
-fn rust_callback(
-    context: &mut ContextWrapper,
-    path: &str,
-    size: i64,
-    kernel_stats: Option<delta_kernel::scan::state::Stats>,
-    dv_info: DvInfo,
-    partition_values: HashMap<String, String>,
-) {
+fn rust_callback(context: &mut ContextWrapper, file: ScanFile) {
```

Comment:
This doesn't work at all. See test failures.

In particular:
- import `ScanFile`
- You can't create a string slice from a `String`, so you need something like:

```rust
    let path = file.path.as_str();
    (context.callback)(
        context.engine_context,
        kernel_string_slice!(path),
        file.size,
        stats.as_ref(),
        &file.dv_info,
        &partition_map,
    );
```

Reply:
Yeah makes sense, fixed. Is there to run these tests locally? I took a look at the GH action but seems like a non-trivial amount of setup.

Reply:
I think you should be able to just run `cargo test` in the ffi crate locally

Reply:
I'm also not seeing a fix? I think you might need to push